File: 92628961-main/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int l=0;
    int s=0;
    int w=1;

    string text = get_string("Text: ");
    int n = strlen(text);
    for(int i = 0; i < n; i++)
    {
       if(text[i] == 33 || text[i] == 46 || text[i] == 63)
       {
         s++;
       }
       if(text[i] == 32)
       {
        w++;
       }
       if (isalpha(text[i]))
       {
        l++;
       }
    }
    float L = 100 * ((float)l / (float)w);
    float S = 100 * ((float)s / (float)w);
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    //L is the average number of letters per 100 words in the text
    //S is the average number of sentences per 100 words in the text.
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: 92628961-main/sentimental-readability/readability.py
import math

l = 0
s = 0
w = 1

text = input("Text: ")
n = len(text)
for i in range(n):
    if text[i] == '!' or text[i] == '.' or text[i] == '?':
        s += 1
    if text[i] == ' ':
        w += 1
    if text[i].isalpha():
        l += 1

L = 100 * (l / w)
S = 100 * (s / w)
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
